app/src/main/java/org/simple/clinic/mobius/MobiusDelegate.kt

```diff
@@ -97,7 +97,8 @@ class MobiusDelegate<M : Parcelable, E, F>(
 
   fun onRestoreInstanceState(bundle: Bundle?): Parcelable? {
     lastKnownModel = bundle?.getParcelable(modelKey) ?: defaultModel
```

Comment:
Can we change this to accept a `Parcelable` and then cast to `Bundle` internally? The delegate can then change the return value of its `onSaveInstanceState` to a `Parcelable` as well. That way, the implementation detail that it saves the state as a `Bundle` can be hidden from the callers and we can avoid the cast at the call site.